runtime/lua/vim/ui/cmdline.lua

```diff
@@ -0,0 +1,102 @@
+local function binary_find(stack, priority, subpriority)
+  if not stack[1] then
+    return 1
+  end
+
+  local a = 1
+  local b = #stack + 1
+
+  while a ~= b do
+    if
+      stack[math.floor((b - a) / 2) + a].priority < priority
+      or (
+        stack[math.floor((b - a) / 2) + a].priority == priority
+        and stack[math.floor((b - a) / 2) + a].subpriority < subpriority
+      )
+    then
+      b = math.floor((b - a) / 2) + a
+    else
+      a = math.floor((b - a) / 2) + a + 1
+    end
+  end
+
+  return a
+end
+
+vim._cmdline_highlighter = function(cmdline)
```

Comment:
```suggestion
local M = {}
M.highlighter = function(cmdline)
```
It's already exposed as `vim._cmdline`.